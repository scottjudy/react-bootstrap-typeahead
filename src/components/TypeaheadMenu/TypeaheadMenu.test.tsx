import React from 'react';

import * as stories from './TypeaheadMenu.stories';

import {
  composeStories,
  generateSnapshots,
  render,
  screen,
} from '../../tests/helpers';

const { CustomChildren, CustomOption, Pagination } = composeStories(stories);

describe('<TypeaheadMenu>', () => {
  generateSnapshots(stories);

  it('renders a custom option', () => {
    render(<CustomOption />);
    expect(screen.getByRole('option')).toHaveTextContent('New selection:');
  });

  it('renders custom new selection text', () => {
    render(<CustomOption newSelectionPrefix="Select new: " />);
    expect(screen.getByRole('option')).toHaveTextContent('Select new:');
  });

  it('renders a pagination item', () => {
    render(<Pagination />);

    const paginationItem = screen.queryByRole('option', {
      name: 'Display additional results...',
    });
    expect(paginationItem).toBeInTheDocument();
  });

  it('renders a custom pagination label', () => {
    const paginationText = 'Show more...';
    render(<Pagination paginationText={paginationText} />);

    expect(
      screen.queryByRole('option', { name: paginationText })
    ).toBeInTheDocument();
  });

  it('renders a custom pagination label component', () => {
    const paginationText = <div>Show more...</div>;
    render(<Pagination paginationText={paginationText} />);

    const item = screen.queryByRole('option', { name: 'Show more...' });
    expect(item).toBeInTheDocument();
    expect(item).toHaveAttribute('aria-label', '');
  });

  it('does not show a paginator when there are no results', () => {
    render(<Pagination options={[]} />);
    expect(
      screen.queryByRole('option', { name: 'Display additional results...' })
    ).not.toBeInTheDocument();
  });

  it('renders custom menu item children', () => {
    render(<CustomChildren />);
    expect(screen.getAllByRole('option')[0]).toHaveTextContent('Population');
  });
});
