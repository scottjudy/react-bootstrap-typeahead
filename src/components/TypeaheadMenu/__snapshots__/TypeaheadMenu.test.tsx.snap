// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TypeaheadMenu> CustomChildren story renders snapshot 1`] = `
<div
  style={
    Object {
      "minHeight": "300px",
    }
  }
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      aria-label="menu-options"
      className="rbt-menu dropdown-menu show"
      id="typeahead-menu"
      onMouseDown={[Function]}
      role="listbox"
      style={
        Object {
          "display": "block",
          "maxHeight": "300px",
          "overflow": "auto",
        }
      }
    >
      <a
        aria-label="Alabama"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-0"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Alabama
        </div>
        <div>
          <small>
            Population: 
            4780127
          </small>
        </div>
      </a>
      <a
        aria-label="Alaska"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-1"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Alaska
        </div>
        <div>
          <small>
            Population: 
            710249
          </small>
        </div>
      </a>
      <a
        aria-label="Arizona"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-2"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Arizona
        </div>
        <div>
          <small>
            Population: 
            6392307
          </small>
        </div>
      </a>
      <a
        aria-label="Arkansas"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-3"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Arkansas
        </div>
        <div>
          <small>
            Population: 
            2915958
          </small>
        </div>
      </a>
      <a
        aria-label="California"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-4"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          California
        </div>
        <div>
          <small>
            Population: 
            37254503
          </small>
        </div>
      </a>
      <a
        aria-label="Colorado"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-5"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Colorado
        </div>
        <div>
          <small>
            Population: 
            5029324
          </small>
        </div>
      </a>
      <a
        aria-label="Connecticut"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-6"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Connecticut
        </div>
        <div>
          <small>
            Population: 
            3574118
          </small>
        </div>
      </a>
      <a
        aria-label="Delaware"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-7"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Delaware
        </div>
        <div>
          <small>
            Population: 
            897936
          </small>
        </div>
      </a>
      <a
        aria-label="Florida"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-8"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Florida
        </div>
        <div>
          <small>
            Population: 
            18804623
          </small>
        </div>
      </a>
      <a
        aria-label="Georgia"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-9"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Georgia
        </div>
        <div>
          <small>
            Population: 
            9688681
          </small>
        </div>
      </a>
      <a
        aria-label="Hawaii"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-10"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Hawaii
        </div>
        <div>
          <small>
            Population: 
            1360301
          </small>
        </div>
      </a>
      <a
        aria-label="Idaho"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-11"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Idaho
        </div>
        <div>
          <small>
            Population: 
            1567652
          </small>
        </div>
      </a>
      <a
        aria-label="Illinois"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-12"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Illinois
        </div>
        <div>
          <small>
            Population: 
            12831549
          </small>
        </div>
      </a>
      <a
        aria-label="Indiana"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-13"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Indiana
        </div>
        <div>
          <small>
            Population: 
            6484229
          </small>
        </div>
      </a>
      <a
        aria-label="Iowa"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-14"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Iowa
        </div>
        <div>
          <small>
            Population: 
            3046869
          </small>
        </div>
      </a>
      <a
        aria-label="Kansas"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-15"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Kansas
        </div>
        <div>
          <small>
            Population: 
            2853132
          </small>
        </div>
      </a>
      <a
        aria-label="Kentucky"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-16"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Kentucky
        </div>
        <div>
          <small>
            Population: 
            4339349
          </small>
        </div>
      </a>
      <a
        aria-label="Louisiana"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-17"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Louisiana
        </div>
        <div>
          <small>
            Population: 
            4533479
          </small>
        </div>
      </a>
      <a
        aria-label="Maine"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-18"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Maine
        </div>
        <div>
          <small>
            Population: 
            1328361
          </small>
        </div>
      </a>
      <a
        aria-label="Maryland"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-19"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Maryland
        </div>
        <div>
          <small>
            Population: 
            5773785
          </small>
        </div>
      </a>
      <a
        aria-label="Massachusetts"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-20"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Massachusetts
        </div>
        <div>
          <small>
            Population: 
            6547817
          </small>
        </div>
      </a>
      <a
        aria-label="Michigan"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-21"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Michigan
        </div>
        <div>
          <small>
            Population: 
            9884129
          </small>
        </div>
      </a>
      <a
        aria-label="Minnesota"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-22"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Minnesota
        </div>
        <div>
          <small>
            Population: 
            5303925
          </small>
        </div>
      </a>
      <a
        aria-label="Mississippi"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-23"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Mississippi
        </div>
        <div>
          <small>
            Population: 
            2968103
          </small>
        </div>
      </a>
      <a
        aria-label="Missouri"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-24"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Missouri
        </div>
        <div>
          <small>
            Population: 
            5988927
          </small>
        </div>
      </a>
      <a
        aria-label="Montana"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-25"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Montana
        </div>
        <div>
          <small>
            Population: 
            989417
          </small>
        </div>
      </a>
      <a
        aria-label="Nebraska"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-26"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Nebraska
        </div>
        <div>
          <small>
            Population: 
            1826341
          </small>
        </div>
      </a>
      <a
        aria-label="Nevada"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-27"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Nevada
        </div>
        <div>
          <small>
            Population: 
            2700691
          </small>
        </div>
      </a>
      <a
        aria-label="New Hampshire"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-28"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          New Hampshire
        </div>
        <div>
          <small>
            Population: 
            1316466
          </small>
        </div>
      </a>
      <a
        aria-label="New Jersey"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-29"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          New Jersey
        </div>
        <div>
          <small>
            Population: 
            8791936
          </small>
        </div>
      </a>
      <a
        aria-label="New Mexico"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-30"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          New Mexico
        </div>
        <div>
          <small>
            Population: 
            2059192
          </small>
        </div>
      </a>
      <a
        aria-label="New York"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-31"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          New York
        </div>
        <div>
          <small>
            Population: 
            19378087
          </small>
        </div>
      </a>
      <a
        aria-label="North Carolina"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-32"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          North Carolina
        </div>
        <div>
          <small>
            Population: 
            9535692
          </small>
        </div>
      </a>
      <a
        aria-label="North Dakota"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-33"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          North Dakota
        </div>
        <div>
          <small>
            Population: 
            672591
          </small>
        </div>
      </a>
      <a
        aria-label="Ohio"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-34"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Ohio
        </div>
        <div>
          <small>
            Population: 
            11536725
          </small>
        </div>
      </a>
      <a
        aria-label="Oklahoma"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-35"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Oklahoma
        </div>
        <div>
          <small>
            Population: 
            3751616
          </small>
        </div>
      </a>
      <a
        aria-label="Oregon"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-36"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Oregon
        </div>
        <div>
          <small>
            Population: 
            3831073
          </small>
        </div>
      </a>
      <a
        aria-label="Pennsylvania"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-37"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Pennsylvania
        </div>
        <div>
          <small>
            Population: 
            12702887
          </small>
        </div>
      </a>
      <a
        aria-label="Rhode Island"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-38"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Rhode Island
        </div>
        <div>
          <small>
            Population: 
            1052931
          </small>
        </div>
      </a>
      <a
        aria-label="South Carolina"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-39"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          South Carolina
        </div>
        <div>
          <small>
            Population: 
            4625401
          </small>
        </div>
      </a>
      <a
        aria-label="South Dakota"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-40"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          South Dakota
        </div>
        <div>
          <small>
            Population: 
            814191
          </small>
        </div>
      </a>
      <a
        aria-label="Tennessee"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-41"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Tennessee
        </div>
        <div>
          <small>
            Population: 
            6346275
          </small>
        </div>
      </a>
      <a
        aria-label="Texas"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-42"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Texas
        </div>
        <div>
          <small>
            Population: 
            25146105
          </small>
        </div>
      </a>
      <a
        aria-label="Utah"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-43"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Utah
        </div>
        <div>
          <small>
            Population: 
            2763888
          </small>
        </div>
      </a>
      <a
        aria-label="Vermont"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-44"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Vermont
        </div>
        <div>
          <small>
            Population: 
            625745
          </small>
        </div>
      </a>
      <a
        aria-label="Virginia"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-45"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Virginia
        </div>
        <div>
          <small>
            Population: 
            8001045
          </small>
        </div>
      </a>
      <a
        aria-label="Washington"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-46"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Washington
        </div>
        <div>
          <small>
            Population: 
            6724543
          </small>
        </div>
      </a>
      <a
        aria-label="West Virginia"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-47"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          West Virginia
        </div>
        <div>
          <small>
            Population: 
            1853011
          </small>
        </div>
      </a>
      <a
        aria-label="Wisconsin"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-48"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Wisconsin
        </div>
        <div>
          <small>
            Population: 
            5687289
          </small>
        </div>
      </a>
      <a
        aria-label="Wyoming"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-49"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        <div>
          Wyoming
        </div>
        <div>
          <small>
            Population: 
            563767
          </small>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`<TypeaheadMenu> CustomOption story renders snapshot 1`] = `
<div
  style={
    Object {
      "minHeight": "300px",
    }
  }
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      aria-label="menu-options"
      className="rbt-menu dropdown-menu show"
      id="typeahead-menu"
      onMouseDown={[Function]}
      role="listbox"
      style={
        Object {
          "display": "block",
          "maxHeight": "300px",
          "overflow": "auto",
        }
      }
    >
      <a
        aria-label="custom option"
        aria-selected={false}
        className="dropdown-item rbt-menu-custom-option"
        href="#"
        id="-item-0"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        New selection: 
        <mark
          className="rbt-highlight-text"
        >
          custom option
        </mark>
      </a>
    </div>
  </div>
</div>
`;

exports[`<TypeaheadMenu> Default story renders snapshot 1`] = `
<div
  style={
    Object {
      "minHeight": "300px",
    }
  }
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      aria-label="menu-options"
      className="rbt-menu dropdown-menu show"
      id="typeahead-menu"
      onMouseDown={[Function]}
      role="listbox"
      style={
        Object {
          "display": "block",
          "maxHeight": "300px",
          "overflow": "auto",
        }
      }
    >
      <a
        aria-label="Alabama"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-0"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Alabama
      </a>
      <a
        aria-label="Alaska"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-1"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Alaska
      </a>
      <a
        aria-label="Arizona"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-2"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Arizona
      </a>
      <a
        aria-label="Arkansas"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-3"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Arkansas
      </a>
      <a
        aria-label="California"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-4"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        California
      </a>
      <a
        aria-label="Colorado"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-5"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Colorado
      </a>
      <a
        aria-label="Connecticut"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-6"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Connecticut
      </a>
      <a
        aria-label="Delaware"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-7"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Delaware
      </a>
      <a
        aria-label="Florida"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-8"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Florida
      </a>
      <a
        aria-label="Georgia"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-9"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Georgia
      </a>
      <a
        aria-label="Hawaii"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-10"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Hawaii
      </a>
      <a
        aria-label="Idaho"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-11"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Idaho
      </a>
      <a
        aria-label="Illinois"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-12"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Illinois
      </a>
      <a
        aria-label="Indiana"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-13"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Indiana
      </a>
      <a
        aria-label="Iowa"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-14"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Iowa
      </a>
      <a
        aria-label="Kansas"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-15"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Kansas
      </a>
      <a
        aria-label="Kentucky"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-16"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Kentucky
      </a>
      <a
        aria-label="Louisiana"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-17"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Louisiana
      </a>
      <a
        aria-label="Maine"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-18"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Maine
      </a>
      <a
        aria-label="Maryland"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-19"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Maryland
      </a>
      <a
        aria-label="Massachusetts"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-20"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Massachusetts
      </a>
      <a
        aria-label="Michigan"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-21"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Michigan
      </a>
      <a
        aria-label="Minnesota"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-22"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Minnesota
      </a>
      <a
        aria-label="Mississippi"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-23"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Mississippi
      </a>
      <a
        aria-label="Missouri"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-24"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Missouri
      </a>
      <a
        aria-label="Montana"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-25"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Montana
      </a>
      <a
        aria-label="Nebraska"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-26"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Nebraska
      </a>
      <a
        aria-label="Nevada"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-27"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Nevada
      </a>
      <a
        aria-label="New Hampshire"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-28"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        New Hampshire
      </a>
      <a
        aria-label="New Jersey"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-29"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        New Jersey
      </a>
      <a
        aria-label="New Mexico"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-30"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        New Mexico
      </a>
      <a
        aria-label="New York"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-31"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        New York
      </a>
      <a
        aria-label="North Carolina"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-32"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        North Carolina
      </a>
      <a
        aria-label="North Dakota"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-33"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        North Dakota
      </a>
      <a
        aria-label="Ohio"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-34"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Ohio
      </a>
      <a
        aria-label="Oklahoma"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-35"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Oklahoma
      </a>
      <a
        aria-label="Oregon"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-36"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Oregon
      </a>
      <a
        aria-label="Pennsylvania"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-37"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Pennsylvania
      </a>
      <a
        aria-label="Rhode Island"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-38"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Rhode Island
      </a>
      <a
        aria-label="South Carolina"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-39"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        South Carolina
      </a>
      <a
        aria-label="South Dakota"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-40"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        South Dakota
      </a>
      <a
        aria-label="Tennessee"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-41"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Tennessee
      </a>
      <a
        aria-label="Texas"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-42"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Texas
      </a>
      <a
        aria-label="Utah"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-43"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Utah
      </a>
      <a
        aria-label="Vermont"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-44"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Vermont
      </a>
      <a
        aria-label="Virginia"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-45"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Virginia
      </a>
      <a
        aria-label="Washington"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-46"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Washington
      </a>
      <a
        aria-label="West Virginia"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-47"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        West Virginia
      </a>
      <a
        aria-label="Wisconsin"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-48"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Wisconsin
      </a>
      <a
        aria-label="Wyoming"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-49"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Wyoming
      </a>
    </div>
  </div>
</div>
`;

exports[`<TypeaheadMenu> Pagination story renders snapshot 1`] = `
<div
  style={
    Object {
      "minHeight": "300px",
    }
  }
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      aria-label="menu-options"
      className="rbt-menu dropdown-menu show"
      id="typeahead-menu"
      onMouseDown={[Function]}
      role="listbox"
      style={
        Object {
          "display": "block",
          "maxHeight": "300px",
          "overflow": "auto",
        }
      }
    >
      <a
        aria-label="Alabama"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-0"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Alabama
      </a>
      <a
        aria-label="Alaska"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-1"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Alaska
      </a>
      <a
        aria-label="Arizona"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-2"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Arizona
      </a>
      <a
        aria-label="Arkansas"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-3"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Arkansas
      </a>
      <a
        aria-label="California"
        aria-selected={false}
        className="dropdown-item"
        href="#"
        id="-item-4"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        California
      </a>
      <div
        className="dropdown-divider"
        role="separator"
      />
      <a
        aria-label="Display additional results..."
        aria-selected={false}
        className="dropdown-item rbt-menu-pagination-option"
        href="#"
        id="-item-5"
        onClick={[Function]}
        onMouseDown={[Function]}
        role="option"
      >
        Display additional results...
      </a>
    </div>
  </div>
</div>
`;
